File: 0540-single-element-in-a-sorted-array/0540-single-element-in-a-sorted-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var singleNonDuplicate = function(nums) {
    let low =0
    let high = nums.length

    while(low<high){
        let mid = Math.floor((high+low)/2)
        if(mid%2==1){
            mid--;
        }
        if(nums[mid] == nums[mid+1]){
            low = mid+2;
        }
        else{
            high = mid;
        }
    }
    return nums[low]


};

File: 0540-single-element-in-a-sorted-array/0540-single-element-in-a-sorted-array.java
class Solution {
    public int singleNonDuplicate(int[] nums) {
          // Violation Process will give me my answer
        int low = 0;
        int high = nums.length-1;
        while(low<high){
            int mid = (low+high)/2;
            if(mid%2==1){
                mid--;
            }
            if(nums[mid]==nums[mid+1]){
                low = mid+2;
            }
            else{
                high = mid;
            }
        }
        return nums[low];
        
    }
}